src/test/scala/scala/tools/refactoring/tests/util/SourceWithMarkerTest.scala

```diff
@@ -306,9 +352,152 @@ class SourceWithMarkerTest {
+    }
+
+    {
+      val src = SourceWithMarker("abc").withMarkerAtLastChar
+      assertTrue(src.withMarkerAtLastChar.moveMarkerBack("a" ~ "b" ~ "c").isDepleted)
+    }
+
+    {
+      val src = SourceWithMarker("abc").withMarkerAtLastChar
+      assertTrue(src.withMarkerAtLastChar.moveMarkerBack("a".atLeastOnce ~ "b".zeroOrMore ~ "c".atLeastOnce).isDepleted)
+    }
+
+    {
+      val src = SourceWithMarker("abc").withMarkerAtLastChar
+      assertTrue(src.withMarkerAtLastChar.moveMarkerBack("a".zeroOrMore ~ "b".atLeastOnce ~ "c".zeroOrMore).isDepleted)
+    }
+
+    {
+      val src = SourceWithMarker("abc").withMarkerAtLastChar
+      assertTrue(src.withMarkerAtLastChar.moveMarkerBack((character('a') | 'c').butNot('d') ~ 'b' ~ 'c').isDepleted)
+    }
+
+    {
+      val src = SourceWithMarker("abc").withMarkerAtLastChar
+      assertTrue(src.withMarkerAtLastChar.moveMarkerBack('a' ~ 'b' ~ 'c').isDepleted)
+    }
+
+    {
+      val src = SourceWithMarker("abc").withMarkerAtLastChar
+      assertTrue(src.withMarkerAtLastChar.moveMarkerBack("abc".butNot("ab" | "bc")).isDepleted)
+    }
+
+    {
+      val src = SourceWithMarker("abc").withMarkerAtLastChar
+      assertTrue(src.withMarkerAtLastChar.moveMarkerBack(idrest).isDepleted)
+    }
+
+    {
+      val src = SourceWithMarker("abc").withMarkerAtLastChar
+      assertTrue(src.withMarkerAtLastChar.moveMarkerBack(varid).isDepleted)
+    }
+  }
+
+  @Test
+  def testWithSimpleExamplesWhereBacktrackingIsNeeded(): Unit = {
+    {
+      val src = SourceWithMarker("aaaab")
+      assertTrue(src.moveMarker('a'.zeroOrMore ~ 'a' ~ 'b').isDepleted)
+      assertTrue(src.moveMarker('a'.zeroOrMore ~ 'a'.nTimes(4) ~ 'b').isDepleted)
+    }
+
+    {
+      val src = SourceWithMarker("aaaab").withMarkerAtLastChar
+      val asOrBs = ('a'.zeroOrMore || 'b'.zeroOrMore)
+      assertTrue(src.moveMarkerBack(asOrBs ~ 'a' ~ 'b').isDepleted)
+      assertTrue(src.moveMarkerBack(asOrBs.zeroOrMore ~ asOrBs.nTimes(4) ~ 'b').isDepleted)
+    }
+
+    {
+      val src = SourceWithMarker("aaab")
+      val mvnt = 'a' ~ 'a'.atLeastOnce
+      assertTrue(src.moveMarker(mvnt ~ (mvnt ~ 'b').backward ~ mvnt ~ 'b').isDepleted)
+    }
+  }
+
+  @Test
+  def testZeroOrMore(): Unit = {
+    {
+      val src = SourceWithMarker("aaa")
+      assertEquals(Seq(3, 2, 1, 0), 'a'.zeroOrMore.compute(src))
+    }
+
+    {
+      val src = SourceWithMarker("")
+      assertTrue(src.applyMovement('a'.zeroOrMore).isDefined)
+    }
+
+    {
+      val src = SourceWithMarker("a")
+      assertTrue(src.moveMarker('a'.zeroOrMore).isDepleted)
+    }
+
+    {
+      val src = SourceWithMarker("aaaa")
+      assertTrue(src.moveMarker('a'.zeroOrMore).isDepleted)
+    }
+
+    {
+      val src = SourceWithMarker("aaaa")
+      assertTrue(src.moveMarker(('a' || 'b').zeroOrMore).isDepleted)
+    }
+
+    {
+      val src = SourceWithMarker("aaaa")
+      assertTrue(src.moveMarker('a'.zeroOrMore.zeroOrMore).isDepleted)
```

Comment:
I agree... `isConsumed` sounds better indeed. A rename refacoring might be in order ;-). However, this might cause some conflicts with PR #144, so maybe we should wait with this until both PRs are merged?